src/main/java/org/folio/rest/impl/InstanceStorageAPI.java

```diff
@@ -75,28 +76,33 @@ public void getInstanceStorageInstances(
     @DefaultValue("10") @Min(1L) @Max(100L) int limit,
     String query,
     @DefaultValue("en") @Pattern(regexp = "[a-zA-Z]{2}") String lang,
-    Map<String, String> okapiHeaders,
+    RoutingContext routingContext, Map<String, String> okapiHeaders,
     Handler<AsyncResult<Response>> asyncResultHandler,
     Context vertxContext) {
 
-    try {
-      PreparedCQL preparedCql = handleCQL(query, limit, offset);
-      PgUtil.getWithOptimizedSql(preparedCql.getTableName(), Instance.class, Instances.class,
-        "title", query, offset, limit,
-        okapiHeaders, vertxContext, GetInstanceStorageInstancesResponse.class, asyncResultHandler);
-    } catch (Exception e) {
-      log.error(e.getMessage(), e);
-      asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
-        GetInstanceStorageInstancesResponse.
-          respond500WithTextPlain(e.getMessage())));
+    if (PgUtil.checkOptimizedCQL(query, "title") != null) { // Until RMB-573 is fixed
+      try {
+        PreparedCQL preparedCql = handleCQL(query, limit, offset);
+        PgUtil.getWithOptimizedSql(preparedCql.getTableName(), Instance.class, Instances.class,
+          "title", query, offset, limit,
+          okapiHeaders, vertxContext, GetInstanceStorageInstancesResponse.class, asyncResultHandler);
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(io.vertx.core.Future.succeededFuture(
+          GetInstanceStorageInstancesResponse.
+            respond500WithTextPlain(e.getMessage())));
+      }
+      return;
     }
```

Comment:
Does this mean that for CQL searches including `title` the response will not be streamed? Or has `PgUtil.getWithOptimizedSql` been changed to stream the response?

Reply:
Correct. PgUtil.getWithOptimizedSql can not be changed as is. It would need new parameters. Created RMB-573 for that work

Reply:
Thank you for raising that additional issue.

In the meantime, I think we need to let folks know about this limitation. Particularly for folks working on [MODINVSTOR-439](https://issues.folio.org/browse/MODINVSTOR-439) and related issues.

Does the `keyword` index also count as including `title` or is that separate and can be streamed?

Reply:
Note.. At this point, only sort by title, is using the special path. So keyword search (other) has no relevance.